src/fiber/fiber.mli

```diff
@@ -228,20 +227,68 @@ end
 with type 'a fiber := 'a t
 
 module Mutex : sig
-  type 'a fiber = 'a t
+  type 'a fiber := 'a t
 
   type t
 
   val create : unit -> t
 
   val with_lock : t -> (unit -> 'a fiber) -> 'a fiber
 end
-with type 'a fiber := 'a t
+
+(** Dealing with cancellation *)
+module Cancel : sig
+  (** Fiber cancellation provides a way to cancel a running fiber. The
+      cancellation mechanism is not builtin the core of fiber and it is not
+      possible to cancel an arbitrary fiber.
+
+      Instead, cancellation is fully explicit. A careful long running fiber may
+      explicitely support cancellation by checking at strategic points if it has
+      been cancelled or not and reacting accordingly. *)
+
+  type 'a fiber := 'a t
+
+  (** A cancellation token. *)
+  type t
+
+  val create : unit -> t
+
+  (** Mark the given cancellation token as cancelled. *)
+  val cancel : t -> unit fiber
+
+  (** [set t f] sets the current cancellation token to [t] while running [f ()]. *)
+  val set : t -> (unit -> 'a fiber) -> 'a fiber
+
+  (** Return whether the current cancellation token has been cancelled. Return
+      [false] if the current fiber doesn't have a cancellation token. *)
+  val cancelled : bool fiber
+
+  type 'a cancelled_or_not =
+    | Cancelled of 'a
+    | Not_cancelled
+
+  (** [with_cancel_handler ~on_cancel f] runs [f ()] with a cancellation
+      handler. If the current cancellation token is cancelled during the
```

Comment:
I struggle parsing sentences saying "cancellation token is cancelled". How about using terms "cancellation" and "request cancellation" instead? Then everything seems to read better:

* `Cancel` -> `Cancellation`
* `Cancel.cancel` -> `Cancellation.request`
* "cancellation token is cancelled" -> "cancellation is requested"
* "checking at strategic points if it has been cancelled" -> "checking at strategic points if cancellation has been requested"

This also seems more faithful to the actual implementation, because `Cancel.cancel` doesn't actually cancel anything, it merely requests a cancellation, which will hopefully eventually be handled.

Reply:
Alright, I'll try to rework the vocabulary